pylibczi/CziFile.py

```diff
@@ -46,20 +50,75 @@ class CziFile(object):
     #   units for the scale in the xml file are not correct (says microns, given in meters)
     scale_units = 1e6
 
-    def __init__(self, czi_filename, metafile_out='', use_pylibczi=True, verbose=False):
-        self.czi_filename = czi_filename
+    # Dims as defined in libCZI
+    #
+    # Z = 1  # The Z-dimension.
+    # C = 2  # The C-dimension ("channel").
+    # T = 3  # The T-dimension ("time").
+    # R = 4  # The R-dimension ("rotation").
+    # S = 5  # The S-dimension ("scene").
+    # I = 6  # The I-dimension ("illumination").
+    # H = 7  # The H-dimension ("phase").
+    # V = 8  # The V-dimension ("view").
+    ####
+    ZISRAW_DIMS = {'Z', 'C', 'T', 'R', 'S', 'I', 'H', 'V', 'B'}
+
+    def __init__(self, czi_filename: types.FileLike, metafile_out: types.PathLike = '', use_pylibczi:bool = True, verbose:bool = False):
+        # Convert to BytesIO (bytestream)
+        self._bytes = self.convert_to_buffer(czi_filename)
+        self.czi_filename = None
         self.metafile_out = metafile_out
         self.czifile_verbose = verbose
 
-        # whether to use czifile or pylibczi for reading the czi file.
-        self.use_pylibczi = use_pylibczi
-        if use_pylibczi:
-            import _pylibczi
-            self.czilib = _pylibczi
+        import _pylibczi
+        self.czilib = _pylibczi
+        self.reader = self.czilib.Reader(self._bytes)
+
+    def dims(self):
+        """
+        Get the dimensions for the opened file from the binary data (not the metadata
+        :return: A dictionary containing Dimension / depth, a file with 3 scenes, 7 time-points and 4 z slices
+         would have {'S': 3, 'T': 7, 'Z':4}
+        """
+        return self.reader.read_dims()
+
+    def is_mosaic(self):
+        return self.reader.is_mosaic()
+
+    @staticmethod
+    def convert_to_buffer(file: types.FileLike) -> io.BufferedIOBase:
```

Comment:
`file` is a built-in object constructor in Python so while it's not a reserved keyword it's kinda iffy to use. I would just switch to `f` or `filelike`